### PR TITLE
Feature/slim 3083 response retrieval

### PR DESCRIPTION
**Pull request by kroesctrl**
Smart metering will first make a call to get the response and afterwards a second call to delete the response data.
For this call we added a seperate wsdl SmartMeteringCommon.wsdl with the deleteResponseData method

### CODE REVIEW

**Comment by kroesctrl** [comment]
https://ci.opensmartgridplatform.org/view/Nightly/job/OSGP_open-smart-grid-platform_nightly/

**Comment by kroesctrl** [comment]
https://ci.opensmartgridplatform.org/view/Nightly/job/OSGP_open-smart-grid-platform_nightly/1297/

**Comment by kroesctrl** [comment]
https://ci.opensmartgridplatform.org/view/Nightly/job/OSGP_open-smart-grid-platform_nightly/1298/

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringbundle/BundleSteps.java [hunk 1]
```diff
@@ -91,6 +92,7 @@
   }
 
   @Autowired private SmartMeteringBundleClient client;
+  @Autowired private ResponseDataRepository responseDataRepository;
```
Comment (harrymiddelburgetall) [inline integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringbundle/BundleSteps.java:95]: Wordt deze ook ergens voor gebruikt?

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.opensmartgridplatform.shared.exceptionhandling.WebServiceSecurityException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ws.soap.client.SoapFaultClientException;
+
+public class CommonResponseDataSteps {
+
+  @Autowired private SmartMeteringCommonClient client;
+
+  @When("^the delete response data request with correlation uid \"([^\"]*)\"$")
+  public void theDeleteResponseDataIsRequested(final String correlationUid)
```
Comment (harrymiddelburgetall) [inline integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java:19]: Method name not in sync with testrule

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    request.setCorrelationUid(correlationUid);
+
+    final DeleteResponseDataResponse response = this.client.sendDeleteResponseDataRequest(request);
+
+    assertThat(response).isNotNull();
+  }
+
+  @When(
+      "^the delete response data request with correlation uid \"([^\"]*)\" should throw SoapFault$")
+  public void theDeleteResponseDataIsRequestedForUnknownUid(final String correlationUid)
```
Comment (harrymiddelburgetall) [inline integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java:32]: Method name not in sync with testrule

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/support/ws/smartmetering/common/SmartMeteringCommonClient.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 Smart Society Services B.V.
```
Comment (harrymiddelburgetall) [inline integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/support/ws/smartmetering/common/SmartMeteringCommonClient.java:2]: 2021. Pasta!

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringCommonEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016 Smart Society Services B.V.
```
Comment (harrymiddelburgetall) [inline osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringCommonEndpoint.java:2]: 2021

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/sm-common.xjb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<!--
+
+    Copyright 2015 Smart Society Services B.V.
```
Comment (harrymiddelburgetall) [inline osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/sm-common.xjb]: 2021

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/sm-common.xjb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+-->
+<jxb:bindings
+  version="2.0"
+  xmlns:jxb="http://java.sun.com/xml/ns/jaxb"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema">
+
+  <jxb:bindings schemaLocation="../schemas/common-ws-smartmetering.xsd">
```
Comment (harrymiddelburgetall) [inline osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/sm-common.xjb:15]: "../schemas/common.xsd"  ?

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringbundle/BundleSteps.java (hunk 1 above)
Comment (kroesctrl) [reply integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringbundle/BundleSteps.java:95]: niet meer idd, verwijderd

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java (hunk 2 above)
Comment (kroesctrl) [reply integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java:19]: Fixed

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java (hunk 3 above)
Comment (kroesctrl) [reply integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringcommon/CommonResponseDataSteps.java:32]: Fixed

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/support/ws/smartmetering/common/SmartMeteringCommonClient.java (hunk 4 above)
Comment (kroesctrl) [reply integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/support/ws/smartmetering/common/SmartMeteringCommonClient.java:2]: Fixed

File: osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringCommonEndpoint.java (hunk 5 above)
Comment (kroesctrl) [reply osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringCommonEndpoint.java:2]: fixed

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/sm-common.xjb (hunk 7 above)
Comment (kroesctrl) [reply osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/sm-common.xjb:15]: Nope, "../schemas/common-ws-smartmetering.xsd" is correct

**Review by harrymiddelburgetall — APPROVED** [review]

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/common.xsd [hunk 8]
```diff
@@ -0,0 +1,788 @@
+<!-- Copyright 2015 Smart Society Services B.V. Licensed under the Apache
```
Comment (kevinsmeets) [inline osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/common.xsd:1]: This XSD file is named `common.xsd`. A while ago a change has been made to ensure more specific names according to a naming convention: https://github.com/OSGP/open-smart-grid-platform/pull/654

I think this name 'common.xsd' goes back to a generic name, which in a way reverts some of the changes made by 654. Can we come up with a different name, preferably following the naming convention?

**Review by kevinsmeets — CHANGES_REQUESTED** [review]

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/common.xsd (hunk 8 above)
Comment (kroesctrl) [reply osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/common.xsd:1]: I will change this file name to base-ws-smartmetering.xsd, as discussed

File: osgp/platform/osgp-adapter-ws-shared/src/main/java/org/opensmartgridplatform/adapter/ws/shared/services/ResponseDataService.java [hunk 9]
```diff
@@ -65,7 +65,45 @@ public ResponseData dequeue(
...
+
+    return responseData;
+  }
 
+  public ResponseData dequeue(final String correlationUid, final ComponentType componentType)
+      throws UnknownCorrelationUidException {
+    final ResponseData responseData = this.get(correlationUid, componentType);
+    this.remove(responseData);
+    return responseData;
+  }
```
[the remark this answers is not in the document]
Comment (kroesctrl) [reply osgp/platform/osgp-adapter-ws-shared/src/main/java/org/opensmartgridplatform/adapter/ws/shared/services/ResponseDataService.java:91]: check

File: integration-tests/cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/common/CommonDeleteResponseData.feature [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+      | OrganizationIdentification | test-org                                 |
+      | DeviceIdentifcation        | TEST1024000000001                        |
+      | MessageType                | REQUEST_ACTUAL_METER_DATA                |
+      | ResultType                 | OK                                       |
+    When the delete response data request with correlation uid "test-org-TEST1024000000001-NOW-666666666"
+    Then the response data record with correlation uid "test-org-TEST1024000000001-NOW-666666666" should be deleted
+
+  Scenario: Delete response data for data that does not exists
+    When the delete response data request with correlation uid "notexist" should throw SoapFault
+    Then the response data record with correlation uid "notexist" should be deleted
```
[the remark this answers is not in the document]
Comment (kroesctrl) [reply integration-tests/cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/common/CommonDeleteResponseData.feature]: Scenario definition has been changed, to the one suggested

**Comment by kroesctrl** [comment]
I made a new PR to support the DCO requirement, for signed commits:
https://github.com/OSGP/open-smart-grid-platform/pull/690

**Pull request closed without merging**